astropy/io/fits/column.py

```diff
@@ -1095,7 +1095,7 @@ def _convert_to_valid_data_type(self, array):
             elif 'L' in format:
                 # boolean needs to be scaled back to storage values ('T', 'F')
                 if array.dtype == np.dtype('bool'):
-                    return np.where(array == False, ord('F'), ord('T'))
+                    return np.where(array == np.bool(False), ord('F'), ord('T'))
```

Comment:
Since this is FITS, I am just going to tag @embray in case he has something to add...